openmetadata-ui/src/main/resources/ui/src/components/DataProductsSelectList/DataProductsSelectList.tsx

```diff
@@ -0,0 +1,187 @@
+/*
+ *  Copyright 2023 Collate.
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+import { Select, Space, Tooltip, Typography } from 'antd';
+import { AxiosError } from 'axios';
+import Loader from 'components/Loader/Loader';
+import { DataProduct } from 'generated/entity/domains/dataProduct';
+import { Paging } from 'generated/type/paging';
+import { debounce } from 'lodash';
+import React, {
+  forwardRef,
+  useCallback,
+  useImperativeHandle,
+  useMemo,
+  useRef,
+  useState,
+} from 'react';
+import { getEntityName } from 'utils/EntityUtils';
+import { tagRender } from 'utils/TagsUtils';
+import { showErrorToast } from 'utils/ToastUtils';
+import {
+  DataProductSelectOption,
+  DataProductsSelectListProps,
+  DataProductsSelectRef,
+} from './DataProductSelectList.interface';
+
+const DataProductsSelectList = forwardRef<
+  DataProductsSelectRef,
+  DataProductsSelectListProps
+>(
+  (
+    {
+      mode,
+      onChange,
+      fetchOptions,
+      debounceTimeout = 800,
+      defaultValue,
+      ...props
+    }: DataProductsSelectListProps,
+    ref
+  ) => {
+    const selectRef = useRef(null);
+    const [isLoading, setIsLoading] = useState(false);
+    const [hasContentLoading, setHasContentLoading] = useState(false);
+    const [options, setOptions] = useState<DataProductSelectOption[]>([]);
+    const [searchValue, setSearchValue] = useState<string>('');
+    const [paging, setPaging] = useState<Paging>({} as Paging);
+    const [currentPage, setCurrentPage] = useState(1);
```

Comment:
Can we use `usePaging` hook here?

Reply:
I would say we wait till the paging bug we found on services is fixed and the approach this.